### PR TITLE
refactor: Use prerender data API

### PR DESCRIPTION
**Pull request by rschristian**
Now that `@preact/preset-vite` supports prerender data, we can remove the hard-coded fallbacks and use the injected prerender data which is likely to be kept closer in-sync. This PR also ditches passing prerender data around with `globalThis` (sorry) for a prerender context provider.

### CODE REVIEW

File: src/components/github-stars.jsx [hunk 1]
```diff
@@ -12,12 +11,11 @@ if (typeof window !== 'undefined') window.githubStars = githubStars;
 
 export default class GithubStars extends Component {
 	state = {
-		stars: localStorageGet('_stars') || fallbackData.preactStargazers
```
Comment (rschristian) [inline src/components/github-stars.jsx:15]: Getting/setting from/to localstorage here seems unnecessary? I don't see any obvious benefit to doing so.

File: src/components/header/gh-version.jsx [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-import { fallbackData, fetchRelease } from '../../lib/github';
+import { injectedPrerenderData, fetchRelease } from '../../lib/github';
 import config from '../../config.json';
 
 export default function ReleaseLink({ ...props }) {
 	const [release, setRelease] = useState({
-		url: globalThis.prerenderPreactReleaseUrl || fallbackData.preactReleaseUrl,
-		version: globalThis.prerenderPreactVersion || fallbackData.preactVersion
+		url: globalThis.prerenderPreactReleaseUrl || injectedPrerenderData.preactReleaseUrl(),
+		version: globalThis.prerenderPreactVersion || injectedPrerenderData.preactVersion()
```
Comment (rschristian) [inline src/components/header/gh-version.jsx:8]: Should probably switch this over to a context yet, rather than passing `globalThis` values around.

File: content/en/index.md [hunk 3]
```diff
@@ -200,7 +200,7 @@ export default class Stars extends Component {
     }
 }
 // --repl-after
-render(&lt;Stars /&gt;, document.getElementById("app"));
+render(&lt;Stars repo="preactjs/preact" /&gt;, document.getElementById("app"));
```
Comment (rschristian) [inline content/en/index.md:203]: Sorry, drive-by issue that I ran into whilst testing:

While our usage example on the home page included the prop, the repl-after here was missing it. This was problematic as it'd attempt to fetch `https://github.com/undefined` if the user tried to open the example in the repl. Must've gotten missed when we added the repl-before & after stuff.

File: src/components/controllers/tutorial/index.jsx [hunk 4]
```diff
@@ -8,12 +8,11 @@ import {
 	useMemo,
 	useCallback
 } from 'preact/hooks';
-import { useLocation, useRoute } from 'preact-iso';
+import { useLocation } from 'preact-iso';
 import { TutorialContext, SolutionContext } from './contexts';
 import { ErrorOverlay } from '../repl/error-overlay';
 import { parseStackTrace } from '../repl/errors';
 import cx from '../../../lib/cx';
-import { InjectPrerenderData } from '../../../lib/prerender-data';
```
Comment (rschristian) [inline src/components/controllers/tutorial/index.jsx:16]: This was a relic of the Preact-CLI setup, `InjectPrerenderData` was linked to a `getPrerenderData` function that hasn't been in use since we switched over. Injecting therefore did nothing.

File: src/index.jsx [hunk 5]
```diff
@@ -35,14 +40,19 @@ export async function prerender() {
...
 		const { version, url } = await (await releaseLambda()).json();
-		globalThis.prerenderPreactVersion = version;
-		globalThis.prerenderPreactReleaseUrl = url;
+		prerenderData.preactVersion = version;
+		prerenderData.preactReleaseURL = url;
+
+		// fetch latest stargazer count
+		const { default: repoLambda } = await import('./lambda/repo.js');
+		const { stargazers_count: stargazersCount } = await (await repoLambda()).json();
+		prerenderData.preactStargazers = stargazersCount;
```
Comment (rschristian) [inline src/index.jsx:49]: As this is very far from crucial (only used in a demo at the very bottom of the home page), I had originally left it out & we just used a hard-coded fallback. Now that this PR introduces a better prerender data setup, however, I felt like it was worth fetching on build & ditching the old hard-coded count.

File: src/components/github-stars.jsx [hunk 6]
```diff
@@ -10,53 +10,42 @@ const formatNumber = num => (num + '').replace(/(\d{3})$/g, ',$1');
 // make available to homepage REPL demo
 if (typeof window !== 'undefined') window.githubStars = githubStars;
 
-export default class GithubStars extends Component {
-	state = {
-		stars: localStorageGet('_stars') || fallbackData.preactStargazers
-	};
+export default function GitHubStars({ user, repo, simple, children }) {
```
Comment (rschristian) [inline src/components/github-stars.jsx:13]: Converted to a function as it simplified consuming the context

File: src/components/header/gh-version.jsx (hunk 2 above)
Comment (rschristian) [reply src/components/header/gh-version.jsx]: DONE

**Review by JoviDeCroock — APPROVED** [review]
